automation/spec/trigger.go

```diff
@@ -54,3 +58,80 @@ func AssertTrigger(t *testing.T, trig trigger.Trigger) {
 		t.Fatalf("json.Unmarshal should emit a `trigger.ErrUnexpectedType` error if the given type does not match the trigger.Type of the Trigger")
 	}
 }
+
+// AssertListener confirms the expected behavior of a trigger.Listener
+// NOTE: this does not confirm behavior of the `Listen` functionality
+// beyond the basic usage of adding a trigger using a `trigger.Source`
+func AssertListener(t *testing.T, listener trigger.Listener, source trigger.Source, activateTrigger func()) {
```

Comment:
To remove the need for the `Bus` method, pass a constructor function as an arg that returns a `Listener` and an `activate` function instead:

```go
// ListenerConstructor creates a trigger listener and function that fires the listener when called
type ListenerConstructor func() (listener trigger.Listener, activate func()

func AssertListener(t *testing.T, constructor ListenerConstructor) {
```

you should also be able to internalize the creation of the workflow being passed as a `source` arg to this test